v2/src/logger.c

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdint.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+#include "c_logging/logger.h"
+#include "c_logging/log_sink.h"
+#include "c_logging/log_sink_console.h"
+#include "c_logging/log_sink_etw.h"
+#include "c_logging/log_context.h"
+
+MU_DEFINE_ENUM_STRINGS_WITHOUT_INVALID(LOG_LEVEL, LOG_LEVEL_VALUES);
+
+const LOG_SINK* log_sinks[] = { &console_log_sink, &etw_log_sink, NULL };
+const char logging_error_message[] = "Error generating log message";
+
+void logger_log(LOG_LEVEL log_level, LOG_CONTEXT_HANDLE log_context, const char* file, const char* func, int line_no, const char* format, ...)
+{
+    uint32_t i;
+    va_list args;
+    va_start(args, format);
+
+    char message[LOG_MAX_MESSAGE_LENGTH];
```

Comment:
there can be a static assert that verifies that the LOG_MESSAGE_SIZE can always accomodate logging_error_message.

Reply:
Yeah this code will have tests checking the bounds.